packages/lisk-transactions/test/utils/validation/validator.js

```diff
@@ -0,0 +1,1019 @@
+			const invalidTransaction = Object.assign({}, validTransaction, {
+				asset: {
+					votes: [
+						'+215b667a32a5cd51a94c9c2046c11fffb08c65748febec099451e3b164452bca',
+						'215b667a32a5cd51a94c9c2046c11fffb08c65748febec099451e3b164452bca',
+					],
+				},
+			});
+			const { valid, errors } = validateTransaction(invalidTransaction);
+			expect(errors[0].dataPath).to.equal('.asset.votes[1]');
+			return expect(valid).to.be.false;
+		});
+
+		it('should validate to be false with errors when asset.votes has non public key', () => {
+			const invalidTransaction = Object.assign({}, validTransaction, {
+				asset: {
+					votes: [
+						'+randome',
+						'-215b667a32a5cd51a94c9c2046c11fffb08c65748febec099451e3b164452bca',
+					],
+				},
+			});
+			const { valid, errors } = validateTransaction(invalidTransaction);
+			expect(errors[0].dataPath).to.equal('.asset.votes[0]');
+			return expect(valid).to.be.false;
+		});
+	});
+
+	describe('transaction type 4', () => {
+		beforeEach(() => {
+			validTransaction = {
+				amount: '0',
+				recipientId: '',
+				senderPublicKey:
+					'a4465fd76c16fcc458448076372abf1912cc5b150663a64dffefe550f96feadd',
+				timestamp: 68249989,
+				type: 4,
+				fee: '1500000000',
+				asset: {
+					multisignature: {
+						min: 2,
+						lifetime: 24,
+						keysgroup: [
+							'+215b667a32a5cd51a94c9c2046c11fffb08c65748febec099451e3b164452bca',
+							'+922fbfdd596fa78269bbcadc67ec2a1cc15fc929a19c462169568d7a3df1a1aa',
+						],
+					},
+				},
+				signature:
+					'b26ce1001190dcad87b80056ce713bbb8c9f3fbeb412b4f04c9c571d31ae7b4f02ad9952f6295f7cfeb625156ec1d29ae9ebf0455907126a16bc61a413b1260f',
+				id: '13972075266355782406',
+			};
+			return Promise.resolve();
+		});
+
+		it('should validate to be true', () => {
+			const { valid, errors } = validateTransaction(validTransaction);
+			expect(errors).to.be.null;
+			return expect(valid).to.be.true;
+		});
+
+		it('should validate to be false with errors when asset.multisignature does not exist', () => {
+			const invalidTransaction = Object.assign({}, validTransaction, {
+				asset: {
+					multisignature: undefined,
+				},
+			});
+			const { valid, errors } = validateTransaction(invalidTransaction);
+			expect(errors[0].params.missingProperty).to.equal('multisignature');
+			return expect(valid).to.be.false;
+		});
+
+		it('should validate to be false with errors when asset.multisignature.min does not exist', () => {
+			const invalidTransaction = Object.assign({}, validTransaction, {
+				asset: {
+					multisignature: {
+						lifetime: 24,
+						keysgroup: [
+							'+215b667a32a5cd51a94c9c2046c11fffb08c65748febec099451e3b164452bca',
+							'+922fbfdd596fa78269bbcadc67ec2a1cc15fc929a19c462169568d7a3df1a1aa',
+						],
+					},
+				},
+			});
+			const { valid, errors } = validateTransaction(invalidTransaction);
+			expect(errors[0].params.missingProperty).to.equal('min');
+			return expect(valid).to.be.false;
+		});
```

Comment:
Is there a general maximum for `min`? What about if `min` is higher than the number of keys in the keysgroup?

Reply:
I think cross field check is over the scope of validator ='(

Reply:
Hmm, but a cross-field check is still something we can statically determine before hitting the API. Maybe it's not something we can use ajv for, but in that case perhaps we should have additional validation depending on the transaction type?